umap/migrations/0023_alter_datalayer_uuid.py

```diff
@@ -0,0 +1,21 @@
+# Generated by Django 5.1.2 on 2024-10-23 14:28
+
+import uuid
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("umap", "0022_add_team"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="datalayer",
+            name="uuid",
+            field=models.UUIDField(
+                default=uuid.uuid4, primary_key=True, serialize=False, unique=True
+            ),
+        ),
+    ]
```

Comment:
I'm actually unsure this migration is needed, but this is what I got after updating the model. I'm curious about your feedback on this @yohanboniface ?

Reply:
I guess because we've set it to editable ? But I'm not sure yet why we needed this.

Reply:
Why we needed to make it editable? Or why the migration is required?

On the former: My understanding is that we need to make it editable, otherwise it's not possible (to my knowledge) to have it provided by the caller on creation. I'll actually try another time to do it without `editable=True`, as it seems better indeed.

Reply:
https://docs.djangoproject.com/en/5.1/ref/models/fields/#editable :)

Reply:
That link says:

> If False, the field will not be displayed in the admin or any other [ModelForm](https://docs.djangoproject.com/en/5.1/topics/forms/modelforms/#django.forms.ModelForm). They are also skipped during [model validation](https://docs.djangoproject.com/en/5.1/ref/models/instances/#validating-objects). Default is True.

(My goal was to actually not skip the validation)

Reply:
Can't we let the DB validate it ? As it will, at the end, no ?

Reply:
I believe it's safer to check the inputs before passing them to the DB 💇